### PR TITLE
Issue 215: Build failed with latest Pravega version

### PR DESCRIPTION
**Pull request by ManishKumarKeshri**
**Changelog description**: 

- **[Issue 4247](https://github.com/pravega/pravega/pull/4243)** in _pravega/pravega_ removed _lombok_ and _netty-all_ from client's dependency list. So, added _lombok_ dependency (as the annotation processor ) in the _pravega-samples._
- **_Updated version_** of **gradle** in _gradle/wrapper/gradle-wrapper.properties_ .

**Purpose of the change**: 

- Fixes: #215  

**What the code does**: 

- Adds lombok dependency. 

**How to verify it:** 

- build must pass

### CODE REVIEW

**Comment by ManishKumarKeshri** [comment]
Hi @sachin-j-joshi @RaulGracia @shrids 

I created a PR on _pravega:dev_. Please review and suggest changes if required.

File: gradle/wrapper/gradle-wrapper.properties [hunk 1]
```diff
@@ -3,4 +3,4 @@ distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
 zipStoreBase=GRADLE_USER_HOME
 zipStorePath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-4.5-bin.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-4.6-bin.zip
```
Comment (shrids) [inline gradle/wrapper/gradle-wrapper.properties]: I did not understand the need for this change. ( `pravega/pravega` uses `gradle-4.10.2-bin.zip` ) . If we should upgrade gradle wrapper then we should upgrade to the latest version.

Comment (ManishKumarKeshri) [reply gradle/wrapper/gradle-wrapper.properties]: Hi @shrids 
Annotation processor is available on Gradle version >= **4.6**. Changed it to the **gradle-4.10.2**. 
Lombok version is **1.18.4**, same as used by _pravega/pravega_.

File: build.gradle [hunk 2]
```diff
@@ -24,6 +24,15 @@ allprojects {
 }
 
 subprojects {
+    if (file("src/main/java").isDirectory()) {
+        apply plugin: 'java'
+    dependencies {
```
Comment (RaulGracia) [inline build.gradle]: Indentation looks incorrect, right?

Comment (ManishKumarKeshri) [reply build.gradle]: Sorry about that. It's good now.

**Comment by RaulGracia** [comment]
@shrids do you have any additional comments?

File: gradle.properties [hunk 3]
```diff
@@ -8,7 +8,7 @@
 #     http://www.apache.org/licenses/LICENSE-2.0
 #
 ### Pravega dependencies
-pravegaVersion=0.6.0-50.076afef-SNAPSHOT
+pravegaVersion=0.6.0-2397.5a60e36-SNAPSHOT
```
Comment (RaulGracia) [inline gradle.properties]: This tag is not publicly available, we need to replace it with the latest artifact published in JFrog repo: https://oss.jfrog.org/artifactory/jfrog-dependencies/io/pravega/

**Pull request merged**